library/src/main/kotlin/com/vinted/actioncable/client/kotlin/Subscriptions.kt

```diff
@@ -21,8 +21,15 @@ class Subscriptions constructor(private val consumer: Consumer) {
      * @param channel Channel to connect
      * @return Subscription instance
      */
-    fun create(channel: Channel): Subscription = Subscription(consumer, channel).also {
-        subscriptions.add(it)
+    fun create(channel: Channel): Subscription {
+        require(channel.identifier !in subscriptions.map { it.identifier }) { "Such subscription already exists" }
```

Comment:
I see the `identifier` contains the name of the channel and other parameters. Different order of parameters leads to a different identifier. Also, created a new `Channel` with the same parameters in the same order leads to a crash. I believe you could get many confusions here.

Reply:
Removed this functionality. User can have multiple subscriptions to the same channel if needed. Added a check ```contains(channel: Channel)``` to check if at least single subscription exists for the channel.